api/handlers.go

```diff
@@ -260,18 +295,23 @@ func (api *API) registerPOST(w http.ResponseWriter, req *http.Request, _ httprou
 		api.WriteError(w, errors.AddContext(err, "failed to validate challenge response"), http.StatusBadRequest)
 		return
 	}
+	// Parse the request's body.
+	var payload credentialsDTO
+	err = json.Unmarshal(body, &payload)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "failed to parse request body"), http.StatusBadRequest)
+		return
+	}
 	// Validate the email address.
-	email := req.PostFormValue("email")
-	e, err := mail.ParseAddress(email)
+	e, err := mail.ParseAddress(payload.Email)
 	if err != nil {
 		api.WriteError(w, errors.New("invalid email provided"), http.StatusBadRequest)
 		return
 	}
 	// Strip any names from the email and leave just the address.
-	email = e.Address
 	// The password is optional.
-	pass := req.PostFormValue("password")
-	u, err := api.staticDB.UserCreatePK(ctx, email, pass, "", pk, database.TierFree)
+	payload.Email = e.Address
```

Comment:
should we update the comment here? it mentions the password, but adjust the email

Reply:
Moved the comment down a bit and expanded to fully explain why it's there - we mention that the password is optional because we're not checking it in any way.